queries.sql

```diff
@@ -9,3 +9,79 @@ SELECT DISTINCT * FROM animals WHERE neutered = true;
 SELECT DISTINCT * FROM animals WHERE name <> 'Gabumon';
 SELECT DISTINCT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+ROLLBACK;
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+ROLLBACK;
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT my_savepoint;
+
+ROLLBACK TO SAVEPOINT my_savepoint;
+
+COMMIT;
+
+ROLLBACK;
+
+BEGIN;
+
+SAVEPOINT my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SAVEPOINT my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+COMMIT;
+
+SELECT COUNT(*) AS animal_count FROM animals;
+
+SELECT COUNT(*) AS never_escaped_count
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight
+FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE EXTRACT(YEAR FROM date_of_birth) BETWEEN 1990 AND 2000
+GROUP BY species;
```

Comment:
- [ ] I noticed that you only attached the result of the query for:

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Kindly attach the screenshots of the results for the remaining queries in the pull request summary. Remember that the screenshots verify that your code was written properly and the results produced are correct. 👍 😎